packages/nimble-components/src/anchor-tabs/index.ts

```diff
@@ -0,0 +1,295 @@
+} from '@microsoft/fast-foundation';
+import { styles } from './styles';
+import { template } from './template';
+import type { AnchorTab } from '../anchor-tab';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-anchor-tabs': AnchorTabs;
+    }
+}
+
+export type TabsOptions = FoundationElementDefinition & StartEndOptions;
+
+/**
+ * A nimble-styled set of anchor tabs
+ */
+export class AnchorTabs extends FoundationElement {
+    /**
+     * The id of the active tab
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: activeid
+     */
+    @attr
+    public activeid = '';
+
+    /**
+     * @internal
+     */
+    @observable
+    public tabs!: HTMLElement[];
+
+    /**
+     * A reference to the active tab
+     * @public
+     */
+    public activetab: HTMLElement | undefined;
+
+    /**
+     * A reference to the tablist div
+     * @internal
+     */
+    public tablist!: HTMLElement;
+
+    private tabIds: string[] = [];
+    private readonly tabAttributeMutationObserver = new MutationObserver(
+        mutations => {
+            mutations.forEach(mutation => {
+                if (
+                    mutation.type === 'attributes'
+                    && mutation.attributeName === 'aria-disabled'
+                ) {
+                    this.setTabs();
+                }
+            });
+        }
+    );
+
+    /**
+     * @internal
+     */
+    public activeidChanged(_oldValue: string, _newValue: string): void {
+        if (this.$fastController.isConnected) {
+            this.setTabs();
+        }
+    }
+
+    /**
+     * @internal
+     */
+    public tabsChanged(): void {
+        if (this.$fastController.isConnected) {
+            this.tabIds = this.getTabIds();
+            this.setTabs();
+        }
+    }
+
+    /**
+     * @internal
+     */
+    public override connectedCallback(): void {
+        super.connectedCallback();
+
+        this.tabIds = this.getTabIds();
+        this.tabAttributeMutationObserver.observe(this, {
+            subtree: true,
+            attributeFilter: ['aria-disabled']
+        });
+    }
+
+    private readonly isDisabledElement = (el: Element): el is HTMLElement => {
+        return el.getAttribute('aria-disabled') === 'true';
+    };
+
+    private readonly isFocusableElement = (el: Element): el is HTMLElement => {
+        return !this.isDisabledElement(el);
+    };
+
+    private readonly setTabs = (): void => {
+        const gridHorizontalProperty = 'gridColumn';
+        const gridVerticalProperty = 'gridRow';
+
+        this.activetab = undefined;
+        this.tabs.forEach((tab: HTMLElement, index: number) => {
+            const tabId: string = this.tabIds[index]!;
+            const isActiveTab = this.activeid === tabId && this.isFocusableElement(tab);
+            tab.setAttribute('id', tabId);
+            tab.setAttribute('aria-selected', isActiveTab ? 'true' : 'false');
+            tab.addEventListener('click', this.handleTabClick);
```

Comment:
I may be misreading this, but it looks like this code might be leaking event listeners. I don't see any calls to `removeEventListener` but it looks like `setTabs` can be called multiple times in the lifetime of the component (e.g. if you set `active-id` repeatedly). And it doesn't look like the `tab` `HTMLElement`s would be recreated, so by my reading we'd be adding the same event listener multiple times.

Reply:
This bit was copied from the FAST implementation, so it looks like they have the same issue. I think the right thing to do is just call `removeEventListener` right before calling `addEventListener`. If it was already registered, we'll remove it, otherwise it is a no-op.